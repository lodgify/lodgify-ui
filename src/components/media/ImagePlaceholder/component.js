import React from 'react';
import { string, oneOfType, number } from 'prop-types';
import classnames from 'classnames';

import { testidFactory } from 'utils/testid';
import { Icon } from 'elements/Icon';

const TEST_PREFIX = 'image-placeholder';
const testid = testidFactory(TEST_PREFIX);

export const Component = ({ className, title, width, height }) => (
  <figure
    className={classnames('image-placeholder', className)}
    data-testid={testid()}
    title={title}
  >
    <Icon name="placeholder" />
  </figure>
);

Component.displayName = 'ImagePlaceholder';
Component.defaultProps = {
  title: null,
  className: '',
};

Component.propTypes = {
  /** Additional ClassName to customize the placeholder. */
  className: string,
  /** The title of the figure that replace the image. */
  title: string
};
