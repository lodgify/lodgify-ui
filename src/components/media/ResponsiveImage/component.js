import React, { PureComponent } from 'react';
import getClassNames from 'classnames';
import PropTypes from 'prop-types';

import {
  IMAGE_NOT_FOUND,
  IMAGE_TITLE,
  IMAGE_WIDGET,
} from 'utils/default-strings';
import { Paragraph } from 'typography/Paragraph';

import { getImageMarkup } from './utils/getImageMarkup';
import { getPlaceholderImageMarkup } from './utils/getPlaceholderImageMarkup';

/**
 * The standard widget for displaying an image.
 */
// eslint-disable-next-line jsdoc/require-jsdoc
export class Component extends PureComponent {
  state = {
    isImageLoaded: false,
    shouldImageLoad: false,
  };

  componentDidMount = () => {
    const { imageUrl, placeholderImageUrl, sizes, srcSet } = this.props;

    if (!!placeholderImageUrl) {
      const fullsizeImage = new Image();

      fullsizeImage.sizes = sizes;
      fullsizeImage.src = imageUrl;
      fullsizeImage.srcset = srcSet;

      this.setState({
        shouldImageLoad: true,
        isImageLoaded: fullsizeImage.complete,
      });

      return;
    }

    this.setState({ shouldImageLoad: true });
  };

  handleImageLoad = () => this.setState({ isImageLoaded: true });

  render = () => {
    const {
      hasRoundedCorners,
      isCircular,
      isFluid,
      label,
      placeholderImageUrl,
    } = this.props;
    const { isImageLoaded, shouldImageLoad } = this.state;

    return (
      <figure
        className={getClassNames('responsive-image', {
          'has-blurred-children': !!placeholderImageUrl && !isImageLoaded,
          'has-placeholder': !!placeholderImageUrl,
          'is-fluid': isFluid,
          'is-rounded': hasRoundedCorners,
          'is-circular': isCircular,
        })}
      >
        {shouldImageLoad && getImageMarkup(this.props, this.handleImageLoad)}
        {!!placeholderImageUrl &&
          getPlaceholderImageMarkup(this.props, isImageLoaded)}
        {label ? <Paragraph>{label}</Paragraph> : null}
      </figure>
    );
  };
}

Component.displayName = 'ResponsiveImage';

Component.defaultProps = {
  alternativeText: IMAGE_WIDGET,
  hasRoundedCorners: false,
  imageHeight: null,
  imageNotFoundLabelText: IMAGE_NOT_FOUND,
  imageTitle: IMAGE_TITLE,
  imageUrl: '',
  imageWidth: null,
  isAvatar: false,
  isCircular: false,
  isFluid: false,
  label: null,
  placeholderImageUrl: undefined,
  sizes: null,
  srcSet: null,
};

Component.propTypes = {
  /** Alternative text to show if the image can't be loaded by the browser. */
  // eslint-disable-next-line react/no-unused-prop-types
  alternativeText: PropTypes.string,
  /** Is the image rounded on the corners. */
  hasRoundedCorners: PropTypes.bool,
  /** The natural height of the image. */
  // eslint-disable-next-line react/no-unused-prop-types
  imageHeight: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  /** The label text for the when the image is not found. */
  // eslint-disable-next-line react/no-unused-prop-types
  imageNotFoundLabelText: PropTypes.string,
  /** Title of the image to show when hovering it on desktop browsers. */
  // eslint-disable-next-line react/no-unused-prop-types
  imageTitle: PropTypes.string,
  /** URL pointing to the image to render. */
  // eslint-disable-next-line react/no-unused-prop-types
  imageUrl: PropTypes.string,
  /** The natural width of the image. */
  // eslint-disable-next-line react/no-unused-prop-types
  imageWidth: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  /** Whether to render the image as an avatar. */
  // eslint-disable-next-line react/no-unused-prop-types
  isAvatar: PropTypes.bool,
  /** Is the image circular. */
  isCircular: PropTypes.bool,
  /** Whether to render fluidly the image or not. */
  // eslint-disable-next-line react/no-unused-prop-types
  isFluid: PropTypes.bool,
  /** A visible label for the image. */
  label: PropTypes.string,
  /** URL pointing to the placeholder image to render. */
  placeholderImageUrl: PropTypes.string,
  /** A list of one or more strings separated by commas indicating a set of source sizes. See [the MDN docs for more information](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/img). */
  sizes: PropTypes.string,
  /** A list of one or more strings separated by commas indicating a set of possible image sources for the user agent to use. See [the MDN docs for more information](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/img). */
  srcSet: PropTypes.string,
};
