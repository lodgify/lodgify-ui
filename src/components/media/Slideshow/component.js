import React, { Fragment, PureComponent } from 'react';
import PropTypes from 'prop-types';
import ImageGallery from 'react-image-gallery';
import { get } from 'lodash';
import getClassNames from 'classnames';

import { Heading } from 'typography/Heading';
import { Paragraph } from 'typography/Paragraph';
import { ICON_NAMES } from 'elements/Icon';

import { adaptImages } from './utils/adaptImages';
import { renderNavButton } from './utils/renderNavButton';

/**
 * A slideshow displays a series of images.
 */
// eslint-disable-next-line jsdoc/require-jsdoc
export class Component extends PureComponent {
  state = {
    index: 0,
  };

  handleSlide = index => this.setState({ index });

  render = () => {
    const {
      hasShadow,
      headingText,
      images,
      isRounded,
      isShowingBulletNavigation,
      isStretched,
    } = this.props;
    const descriptionText = get(images, [this.state.index, 'descriptionText']);

    return (
      <Fragment>
        {headingText && <Heading>{headingText}</Heading>}
        {descriptionText && <Paragraph>{descriptionText}</Paragraph>}
        <ImageGallery
          additionalClass={getClassNames('', {
            'fit-height': isStretched,
            'no-shadow': !hasShadow,
            'no-border-radius': !isRounded,
          })}
          // Note: styles for the pagination controls
          // live in `styles/semantic/themes/livingstone/collections/menu.*`
          items={adaptImages(images)}
          lazyLoad
          onSlide={this.handleSlide}
          renderLeftNav={renderNavButton(ICON_NAMES.CHEVRON_LEFT)}
          renderRightNav={renderNavButton(ICON_NAMES.CHEVRON_RIGHT)}
          showBullets={isShowingBulletNavigation}
          showFullscreenButton={false}
          showPlayButton={false}
          showThumbnails={false}
        />
      </Fragment>
    );
  };
}

Component.displayName = 'Slideshow';

Component.defaultProps = {
  hasShadow: true,
  headingText: null,
  isRounded: true,
  isShowingBulletNavigation: true,
  isStretched: false,
};

Component.propTypes = {
  /** Has the slideshow got a drop shadow  */
  hasShadow: PropTypes.bool,
  /** The text to display as a heading at the top of the slideshow. */
  headingText: PropTypes.string,
  /**  The images to display.  */
  images: PropTypes.arrayOf(
    PropTypes.shape({
      /** Alternative text to show if the image can't be loaded by the browser. */
      alternativeText: PropTypes.string,
      /** A description of the image to show above the slideshow when the image is showing. */
      descriptionText: PropTypes.string,
      /** A list of one or more strings separated by commas indicating a set of source sizes. See [the MDN docs for more information](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/img). */
      sizes: PropTypes.string,
      /** A list of one or more strings separated by commas indicating a set of possible image sources for the user agent to use. See [the MDN docs for more information](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/img). */
      srcSet: PropTypes.string,
      /** Title of the image to show when hovering over it on desktop browsers. */
      title: PropTypes.string,
      /** URL pointing to the image to display. */
      url: PropTypes.string.isRequired,
    })
  ).isRequired,
  /** Is the slideshow formatted to have rounded corners */
  isRounded: PropTypes.bool,
  /** Is the slideshow displaying the bullet navigation */
  isShowingBulletNavigation: PropTypes.bool,
  /** Is the slideshow fitting the height of it's container */
  isStretched: PropTypes.bool,
};
