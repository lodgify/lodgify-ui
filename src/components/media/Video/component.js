import ReactPlayer from 'react-player';
import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import DOMPurify from 'dompurify';
import isValidUrl from 'is-url';
import isValidHTML from 'is-html';
import getClassNames from 'classnames';

import { getPlayerCss } from './utils/getPlayerCss';
import { getReactPlayerProps } from './utils/getReactPlayerProps';
import { logWarning } from './utils/logWarning';

/**
 * The Video widget. It allows the consumer to render videos given a URL or
 * an embeddable HTML snippet.
 * @extends {React.PureComponent}
 * @returns {Object}
 */
export class Component extends PureComponent {
  state = {
    cleanHTMLString: null,
  };

  componentDidMount = () => {
    const { videoSource } = this.props;

    isValidHTML(videoSource) &&
      this.setState({
        // `DOMPurify.sanitize` is in `componentDidMount` so that it
        // is not called during server side rendering.
        // Reason: DOMPurify depends on browser features.
        cleanHTMLString: DOMPurify.sanitize(videoSource, {
          SANITIZE_DOM: true,
          ALLOWED_TAGS: ['iframe', 'object', 'video', 'audio'],
        }),
      });
  };

  render() {
    const { videoSource, isResponsive, width, height } = this.props;

    const playerWrapperProps = {
      className: getClassNames('video', 'player-wrapper', {
        'is-url': isValidUrl(videoSource),
        'is-html': isValidHTML(videoSource),
        'is-responsive': isResponsive,
      }),
      style: getPlayerCss(isResponsive, width, height),
    };

    switch (true) {
      case isValidUrl(videoSource):
        return (
          <div {...playerWrapperProps}>
            <ReactPlayer {...getReactPlayerProps(isResponsive, videoSource)} />
          </div>
        );
      case isValidHTML(videoSource):
        return (
          <div
            {...playerWrapperProps}
            dangerouslySetInnerHTML={{
              __html: this.state.cleanHTMLString,
            }}
          />
        );
      default:
        logWarning(videoSource);
        return null;
    }
  }
}

Component.displayName = 'Video';

Component.defaultProps = {
  height: 315,
  isResponsive: false,
  videoSource: null,
  width: 560,
};

Component.propTypes = {
  /** The height of the video to determine the aspect ratio of the video */
  height: PropTypes.number,
  /** Should the video be responsive */
  isResponsive: PropTypes.bool,
  /** The string that will be used to build the video player.
   *  A URL pointing to a Youtube, Vimeo, ... video is valid.
   *  A embeddable HTML video snippet is also valid (iframes, video, audio, objects)
   *  An invalid input throws an error.
   */
  videoSource: PropTypes.string,
  /** The width of the component to determine aspect ratio */
  width: PropTypes.number,
};
