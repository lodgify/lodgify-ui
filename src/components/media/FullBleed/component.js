import React from 'react';
import PropTypes from 'prop-types';
import getClassNames from 'classnames';
import { Segment } from 'semantic-ui-react';
import { size } from 'lodash';

import { ResponsiveImage } from 'media/ResponsiveImage';

import { DEFAULT_BOTTOM_OFFSET } from './constants';

/**
 * A full bleed takes up the width of its containing element
 * and the height of the viewport.
 */
// eslint-disable-next-line jsdoc/require-jsdoc
export const Component = ({
  bottomOffset,
  children,
  hasGradient,
  imageUrl,
  imageHeight,
  imageWidth,
  placeholderImageUrl,
  sizes,
  srcSet,
}) => (
  <Segment
    className={getClassNames('full-bleed', {
      'has-gradient': hasGradient,
      'has-children': size(children) > 0,
    })}
    style={{
      '--full-bleed-bottom-offset': bottomOffset,
    }}
    vertical
  >
    <ResponsiveImage
      imageHeight={imageHeight}
      imageUrl={imageUrl}
      imageWidth={imageWidth}
      isFluid
      placeholderImageUrl={placeholderImageUrl}
      sizes={sizes}
      srcSet={srcSet}
    />
    {children}
  </Segment>
);

Component.displayName = 'FullBleed';

Component.defaultProps = {
  bottomOffset: DEFAULT_BOTTOM_OFFSET,
  children: null,
  hasGradient: false,
  imageHeight: undefined,
  imageUrl: null,
  imageWidth: undefined,
  placeholderImageUrl: null,
  sizes: null,
  srcSet: null,
};

Component.propTypes = {
  /** Reduce the height of the FullBleed with an offset, supports CSS dimensions. */
  bottomOffset: PropTypes.string,
  /** The children to render inside the full bleed. */
  children: PropTypes.node,
  /** Is there a gradient overlaying the full bleed.  */
  hasGradient: PropTypes.bool,
  /** The natural height of the image. */
  imageHeight: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  /** URL pointing to the image to render. */
  imageUrl: PropTypes.string,
  /** The natural width of the image. */
  imageWidth: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  /** URL pointing to the placeholder image to render. */
  placeholderImageUrl: PropTypes.string,
  /** A list of one or more strings separated by commas indicating a set of source sizes for the image. See [the MDN docs for more information](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/img). */
  sizes: PropTypes.string,
  /** A list of one or more strings separated by commas indicating a set of possible image sources for the user agent to use for the image. See [the MDN docs for more information](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/img). */
  srcSet: PropTypes.string,
};
