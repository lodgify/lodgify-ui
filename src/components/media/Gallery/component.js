import React, { Fragment } from 'react';
import PropTypes from 'prop-types';

import { Modal } from 'elements/Modal';
import { HorizontalGutters } from 'layout/HorizontalGutters';
import { Divider } from 'elements/Divider';
import { Grid } from 'layout/Grid';
import { GridRow } from 'layout/GridRow';
import { GridColumn } from 'layout/GridColumn';
import { buildKeyFromStrings } from 'utils/build-key-from-strings';
import { Heading } from 'typography/Heading';
import { ResponsiveImage } from 'media/ResponsiveImage';

/**
 * A gallery displays a collection of images in a modal.
 */
// eslint-disable-next-line jsdoc/require-jsdoc
export const Component = ({ heading, images, trigger }) => (
  <Modal isFullscreen trigger={trigger}>
    <HorizontalGutters>
      <Divider />
      {heading && (
        <Fragment>
          {heading}
          <Divider hasLine />
        </Fragment>
      )}
      <Grid columns={2} stackable>
        <GridRow>
          {images.map(({ label, url, ...otherProps }, index) => (
            <GridColumn key={buildKeyFromStrings(label, index)}>
              <Heading size="small">{label}</Heading>
              <ResponsiveImage imageUrl={url} {...otherProps} />
              <Divider />
            </GridColumn>
          ))}
        </GridRow>
      </Grid>
      <Divider size="large" />
    </HorizontalGutters>
  </Modal>
);

Component.displayName = 'Gallery';

Component.defaultProps = {
  heading: null,
};

Component.propTypes = {
  /** The content to display as a heading at the top of the gallery. */
  heading: PropTypes.node,
  /** The images to display. See [the `ResponsiveImage` component for valid props](#/Media/ResponsiveImage).  */
  images: PropTypes.arrayOf(PropTypes.object).isRequired,
  /** The element to be clicked to display the gallery. */
  trigger: PropTypes.node.isRequired,
};
