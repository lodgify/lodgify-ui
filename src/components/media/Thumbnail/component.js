import React from 'react';
import PropTypes from 'prop-types';
import getClassNames from 'classnames';

import { ResponsiveImage } from 'media/ResponsiveImage';
import { Paragraph } from 'typography/Paragraph';

/**
 * A thumbnail displays a small image.
 */
// eslint-disable-next-line jsdoc/require-jsdoc
export const Component = ({
  className,
  hasRoundedCorners,
  imageAlternativeText,
  imageSizes,
  imageSrcSet,
  imageUrl,
  isCircular,
  isSquare,
  label,
  size,
}) => (
  <div
    className={getClassNames('ui', 'thumbnail', size, className, {
      'is-square': isSquare,
      'is-circular': isCircular,
    })}
  >
    <ResponsiveImage
      alternativeText={imageAlternativeText}
      hasRoundedCorners={hasRoundedCorners}
      imageUrl={imageUrl}
      isCircular={isCircular}
      isSquare={isSquare}
      sizes={imageSizes}
      srcSet={imageSrcSet}
    />
    {!!label && <Paragraph>{label}</Paragraph>}
  </div>
);

Component.displayName = 'Thumbnail';

Component.defaultProps = {
  className: '',
  hasRoundedCorners: false,
  imageAlternativeText: undefined,
  imageSizes: undefined,
  imageSrcSet: undefined,
  isCircular: false,
  isSquare: false,
  label: null,
  size: null,
};

Component.propTypes = {
  /**
   * Custom class name.
   * Provided by `ShowOnMobile` and `ShowOnDesktop` so ignored in the styleguide.
   * @ignore
   */
  className: PropTypes.string,
  /** Is the thumbnail rounded on the corners */
  hasRoundedCorners: PropTypes.bool,
  /** Text to help visually impaired users understand the content of the image. */
  imageAlternativeText: PropTypes.string,
  /** A list of one or more strings separated by commas indicating a set of source sizes. See [the MDN docs for more information](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/img). */
  imageSizes: PropTypes.string,
  /** A list of one or more strings separated by commas indicating a set of possible image sources for the user agent to use. See [the MDN docs for more information](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/img). */
  imageSrcSet: PropTypes.string,
  /** URL pointing to the image to render. */
  imageUrl: PropTypes.string.isRequired,
  /** Is the thumbnail circular */
  isCircular: PropTypes.bool,
  /** Is the thumbnail square */
  isSquare: PropTypes.bool,
  /** A visible label for the thumbnail */
  label: PropTypes.string,
  /** The size of the thumbnail */
  size: PropTypes.oneOf(['small', 'large', 'huge']),
};
