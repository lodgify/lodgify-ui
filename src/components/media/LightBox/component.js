import React from 'react';
import PropTypes from 'prop-types';

import { Modal } from 'elements/Modal';
import { Slideshow } from 'media/Slideshow';
import { testidFactory } from 'utils/testid';

const TEST_ID_PREFIX = 'lightbox';

const testid = testidFactory(TEST_ID_PREFIX);

/**
 * A LightBox which displays an image or a series of images in a modal.
 */
// eslint-disable-next-line jsdoc/require-jsdoc
export const Component = ({ slideShowImages, trigger }) => {
  const areThumbnailsActive = slideShowImages.length > 1;

  return (
    <Modal
      className="is-lightbox"
      data-testid={testid()}
      hasCloseIcon
      isFullscreen
      trigger={trigger}
    >
      <Slideshow
        hasShadow={false}
        images={slideShowImages}
        isRounded={false}
        isShowingBulletNavigation={false}
        isShowingIndex={areThumbnailsActive}
        isShowingThumbnails={areThumbnailsActive}
        thumbnails={slideShowImages}
      />
    </Modal>
  );
};

Component.displayName = 'LightBox';

Component.defaultProps = {
  trigger: null,
};

Component.propTypes = {
  /**  The images to display inside the Slideshow. */
  slideShowImages: PropTypes.arrayOf(
    PropTypes.shape({
      /** A description of the image to show above the slideshow when the image is showing. */
      descriptionText: PropTypes.string,
      /** A list of one or more strings separated by commas indicating a set of source sizes. See [the MDN docs for more information](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/img). */
      sizes: PropTypes.string,
      /** A list of one or more strings separated by commas indicating a set of possible image sources for the user agent to use. See [the MDN docs for more information](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/img). */
      srcSet: PropTypes.string,
      /** Title of the image to show when hovering over it on desktop browsers. */
      title: PropTypes.string,
      /** URL pointing to the image to display. */
      url: PropTypes.string.isRequired,
    })
  ).isRequired,
  /** The element to be clicked to display the modal. */
  trigger: PropTypes.node,
};
