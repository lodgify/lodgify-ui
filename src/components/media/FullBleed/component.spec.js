import React from 'react';
import { shallow } from 'enzyme';
import { Segment } from 'semantic-ui-react';
import {
  expectComponentToBe,
  expectComponentToHaveDisplayName,
  expectComponentToHaveProps,
} from '@lodgify/enzyme-jest-expect-helpers';

import { getBackgroundImageUrl } from 'utils/get-background-image-url';

import { Component as FullBleed } from './component';

const props = {
  children: '🚸',
  imageUrl: '🐱🐱',
};

const getFullBleed = extraProps =>
  shallow(<FullBleed {...props} {...extraProps} />);
const getSegment = extraProps => getFullBleed(extraProps).find(Segment);

describe('<FullBleed />', () => {
  it('should render a single Semantic UI `Segment` component', () => {
    const wrapper = getFullBleed();

    expectComponentToBe(wrapper, Segment);
  });

  describe('the `Segment` component', () => {
    it('should have the right props', () => {
      const wrapper = getSegment();

      expectComponentToHaveProps(wrapper, {
        children: props.children,
        className: 'full-bleed',
        style: { backgroundImage: getBackgroundImageUrl(props.imageUrl) },
        vertical: true,
      });
    });
  });

  describe('if `hasGradient` is passed', () => {
    it('should have the correct `props.className`', () => {
      const wrapper = getSegment({ hasGradient: true });

      expectComponentToHaveProps(wrapper, {
        className: 'full-bleed has-gradient',
      });
    });
  });

  describe('if `props.className` is passed', () => {
    it('should have the correct `props.className`', () => {
      const wrapper = getSegment({ className: '🏛' });

      expectComponentToHaveProps(wrapper, {
        className: 'full-bleed 🏛',
      });
    });
  });

  it('should have `displayName` `FullBleed`', () => {
    expectComponentToHaveDisplayName(FullBleed, 'FullBleed');
  });
});
