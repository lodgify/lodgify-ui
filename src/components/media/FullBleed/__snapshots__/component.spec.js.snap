// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FullBleed /> by default should have the right structure 1`] = `
<FullBleed
  bottomOffset="0px"
  hasGradient={false}
  imageUrl="üê±üê±"
  placeholderImageUrl={null}
  sizes={null}
  srcSet={null}
>
  <Segment
    className="full-bleed"
    style={
      Object {
        "--full-bleed-bottom-offset": "0px",
      }
    }
    vertical={true}
  >
    <div
      className="ui vertical segment full-bleed"
      style={
        Object {
          "--full-bleed-bottom-offset": "0px",
        }
      }
    >
      <ResponsiveImage
        alternativeText="Image Widget"
        hasRoundedCorners={false}
        imageHeight={null}
        imageNotFoundLabelText="Image not found!"
        imageTitle="Image Title"
        imageUrl="üê±üê±"
        imageWidth={null}
        isAvatar={false}
        isCircular={false}
        isFluid={true}
        label={null}
        placeholderImageUrl={null}
        sizes={null}
        srcSet={null}
      >
        <figure
          className="responsive-image is-fluid"
        >
          <Image
            alt="Image Widget"
            as="img"
            avatar={false}
            fluid={true}
            onLoad={[Function]}
            sizes={null}
            src="üê±üê±"
            srcSet={null}
            title="Image Title"
            ui={true}
          >
            <img
              alt="Image Widget"
              className="ui fluid image"
              onLoad={[Function]}
              sizes={null}
              src="üê±üê±"
              srcSet={null}
              title="Image Title"
            />
          </Image>
        </figure>
      </ResponsiveImage>
    </div>
  </Segment>
</FullBleed>
`;

exports[`<FullBleed /> if \`hasGradient\` is passed should have the right structure 1`] = `
<FullBleed
  bottomOffset="0px"
  hasGradient={true}
  imageUrl="üê±üê±"
  placeholderImageUrl={null}
  sizes={null}
  srcSet={null}
>
  <Segment
    className="full-bleed has-gradient"
    style={
      Object {
        "--full-bleed-bottom-offset": "0px",
      }
    }
    vertical={true}
  >
    <div
      className="ui vertical segment full-bleed has-gradient"
      style={
        Object {
          "--full-bleed-bottom-offset": "0px",
        }
      }
    >
      <ResponsiveImage
        alternativeText="Image Widget"
        hasRoundedCorners={false}
        imageHeight={null}
        imageNotFoundLabelText="Image not found!"
        imageTitle="Image Title"
        imageUrl="üê±üê±"
        imageWidth={null}
        isAvatar={false}
        isCircular={false}
        isFluid={true}
        label={null}
        placeholderImageUrl={null}
        sizes={null}
        srcSet={null}
      >
        <figure
          className="responsive-image is-fluid"
        >
          <Image
            alt="Image Widget"
            as="img"
            avatar={false}
            fluid={true}
            onLoad={[Function]}
            sizes={null}
            src="üê±üê±"
            srcSet={null}
            title="Image Title"
            ui={true}
          >
            <img
              alt="Image Widget"
              className="ui fluid image"
              onLoad={[Function]}
              sizes={null}
              src="üê±üê±"
              srcSet={null}
              title="Image Title"
            />
          </Image>
        </figure>
      </ResponsiveImage>
    </div>
  </Segment>
</FullBleed>
`;

exports[`<FullBleed /> if \`props.children > 0\` should have the right structure 1`] = `
<FullBleed
  bottomOffset="0px"
  hasGradient={false}
  imageUrl="üê±üê±"
  placeholderImageUrl={null}
  sizes={null}
  srcSet={null}
>
  <Segment
    className="full-bleed has-children"
    style={
      Object {
        "--full-bleed-bottom-offset": "0px",
      }
    }
    vertical={true}
  >
    <div
      className="ui vertical segment full-bleed has-children"
      style={
        Object {
          "--full-bleed-bottom-offset": "0px",
        }
      }
    >
      <ResponsiveImage
        alternativeText="Image Widget"
        hasRoundedCorners={false}
        imageHeight={null}
        imageNotFoundLabelText="Image not found!"
        imageTitle="Image Title"
        imageUrl="üê±üê±"
        imageWidth={null}
        isAvatar={false}
        isCircular={false}
        isFluid={true}
        label={null}
        placeholderImageUrl={null}
        sizes={null}
        srcSet={null}
      >
        <figure
          className="responsive-image is-fluid"
        >
          <Image
            alt="Image Widget"
            as="img"
            avatar={false}
            fluid={true}
            onLoad={[Function]}
            sizes={null}
            src="üê±üê±"
            srcSet={null}
            title="Image Title"
            ui={true}
          >
            <img
              alt="Image Widget"
              className="ui fluid image"
              onLoad={[Function]}
              sizes={null}
              src="üê±üê±"
              srcSet={null}
              title="Image Title"
            />
          </Image>
        </figure>
      </ResponsiveImage>
      üèõ
    </div>
  </Segment>
</FullBleed>
`;
